core/src/test/java/org/infinispan/functional/decorators/FunctionalAdvancedCache.java

```diff
@@ -643,4 +645,186 @@ public void removeListener(Object listener) {
       }
    }
 
+   private static final class SetAsCacheSet<E> implements CacheSet<E> {
```

Comment:
You should be able to use `AbstractDelegatingSet` as the base class here and then only have to implement the CacheSet specific override methods.